src/extensibility/node/ExtensionManagerDomain.js

```diff
@@ -80,6 +81,54 @@ function _removeFailedInstallation(installDirectory) {
 }
 
 /**
+ * Private function to run 'npm install --production' command in the extension directory.
+ *
+ * @param {string} installDirectory Directory to remove
+ * @param {function} callback NodeJS style callback to call after finish
+ */
+function _performNpmInstall(installDirectory, callback) {
+    var npmPath = path.resolve(path.dirname(require.resolve("npm")), "..", "bin", "npm-cli.js");
+    var args = [npmPath, 'install', '--production'];
+
+    console.log("running npm install --production in " + installDirectory);
+
+    var child = spawn(process.execPath, args, { cwd: installDirectory });
+
+    child.on("error", function (err) {
+        return callback(err);
+    });
+
+    var stdout = [];
+    child.stdout.addListener("data", function (buffer) {
+        stdout.push(buffer);
+    });
+
+    var stderr = [];
+    child.stderr.addListener("data", function (buffer) {
+        stderr.push(buffer);
+    });
+
+    var exitCode;
+    child.addListener("exit", function (code) {
+        exitCode = code;
+    });
+
+    child.addListener("close", function () {
+        stderr = Buffer.concat(stderr);
+        stdout = Buffer.concat(stdout);
+        if (exitCode > 0) {
```

Comment:
I assume `exit` is always called and `close` is always called after it.
Otherwise is better to initialize `exitCode`.